rustls/src/time_provider.rs

```diff
@@ -0,0 +1,50 @@
+//! no-std time provider
+
+use alloc::sync::Arc;
+use core::fmt::Debug;
+
+use pki_types::UnixTime;
+
+/// An object that provides the current time.
+///
+/// This is used to, for example, check if a certificate has expired during certificate validation,
+/// or to check the age of a ticket.
+#[derive(Clone, Debug)]
+pub struct TimeProvider {
```

Comment:
What value does this wrapper provide? IMO it would be more clear to avoid this wrapper and use `Arc<dyn GetCurrentTime>` directly.